### PR TITLE
test(e2e): refactor collect function

### PR DESCRIPTION
**Pull request by michaelbeaumont**
There was some weirdness going on with faking responses to match the echo format.

This PR also adds the ability to get both the response body as well as the response headers from `curl`.

It removes `-tt` from SSH calls because it had the effect of outputting `stderr` on `stdout`. That required fixes everywhere we use raw `curl`.

UNFORTUNATELY, the above breaks signal handling. It's very difficult to get separate streams + signal handling apparently.

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: test/framework/client/collect.go [hunk 1]
```diff
@@ -126,8 +126,7 @@ func OutputFormat(format string) CollectResponsesOptsFn {
 		// parse the format they specified.
 		opts.Flags = append(opts.Flags,
 			"--silent",
-			"--output", os.DevNull,
-			"--write-out", opts.ShellEscaped(format),
+			"--write-out", format,
```
Comment (lobkovilya) [inline test/framework/client/collect.go:129]: Why you removed `ShellEscaped`?

Comment (michaelbeaumont) [reply test/framework/client/collect.go]: Fixed it by changing the ordering of the options

**Comment by jakubdyszkiewicz** [comment]
a test is down it seems? Please run it with full test matrix before merging just to double check that everything is fine.

**Review by jakubdyszkiewicz — APPROVED** [review]

**Comment by michaelbeaumont** [comment]
Something has definitely become flaky with:

```
[It] Test Projected Service Account Token on Universal should connect to restarted control plane with new token without dp restart [job-0]
```

:thinking:

**Comment by michaelbeaumont** [comment]
Closing this as it was a mostly unsuccessful refactor

**Pull request closed without merging**